lib/types/array/index.js

```diff
@@ -49,7 +49,14 @@ internals.Array = class extends Any {
         if (typeof value === 'string' &&
             options.convert) {
 
-            internals.safeParse(value, result);
+            if (value.length > 1 &&
+                (value[0] === '[' || /^\s*\[/.test(value))) {
```

Comment:
Does it make sense to check that the opening `[` is balanced at least with a `]` at the last character (accounting for trailing whitespace like your `[` regex does)?


Or is this simply a quick and dirty way to guard against Objects the way that `Array.isArray()` was doing?

Reply:
You could do that, but I doubt there are many cases where we will see a string that starts with `[`, and doesn't end with an `]`. As such a test would incur extra costs whenever the regex is reached, I suspect the one I do is more suitable.

I'm just trying to screen obviously non-array strings, which could be encountered in an `alternatives()` check. Eg. for `Joi.alternatives().try([Joi.array(), Joi.number()])`.